app/models/request.rb

```diff
@@ -1,16 +1,81 @@
-class Request < ActiveRecord::Base
-  belongs_to :application
-  belongs_to :event_source, counter_cache: true
-  belongs_to :raw_event
-  belongs_to :scenario
-
-  has_many :events, dependent: :destroy
-
-  validates :application, presence: true
-  validates :event_source, presence: true
-  validates :raw_event, presence: true
-  validates :controller, presence: true
-  validates :action, presence: true
-  validates :duration, presence: true
-  validates :scenario, presence: true
+class Request
+  include ActiveModel::Model
+
+  attr_accessor :application, :scenario, :event_source, :controller, :action,
+    :timestamp, :duration, :started_at, :finished_at, :raw_event
+
+  class << self
+    def sum_duration(event_source:, from:, to:)
+      from ||= Time.now - 1.week
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.